### ISSUE TITLE
Build issue prevents Razor from working in functions v2

### ISSUE THREAD

**Issue by Doddler** [Azure/azure-functions-host#4132]
I've been experiencing a number of issues trying to get the Razor View Engine working in Azure Functions (v2 using 1.0.24 of the c# sdk and 2.4.317 of the cli tools), and have narrowed down the issues to what is likely an issue or number of issues in the build process. I've created a repo project that contains both a functions project and a console project calling the same code from a class library to different results: https://github.com/Doddler/FunctionsRazorTest

The first issue is, when attempting to compile a razor view, you'll encounter a wall of errors suggesting basic predefined types (string, object, type, etc) are not defined. This can be encountered in the console app as well in certain circumstances, but the solution to that is to set this property in your csproject:

```<PreserveCompilationContext>true</PreserveCompilationContext>```

As I understand it, this option generates additional details in the AssemblyName.deps.json that is used by Razor. This doesn't solve the issue in the Functions project, but in testing I found that I could fix the errors by moving the .deps.json file into the bin directory. The rendering still fails with a different error (Cannot find compilation library location for package), and I've yet to solve this one, but it's clear that the file isn't where it's being looked for.

To work around the rendering issue, I decided to explore precompiled views using a class library, but encountered a very similar issue. A razor class library generates two files, Libraryname.dll and Libraryname.Views.dll. During the build process, the latter file does not get moved into the bin directory and hence not visible to the function app. Manually moving the views dll into the bin directory and re-running results in the view finally being rendered correctly.

To sum up, some files aren't being found because they aren't being moved into the bin directory on compile.

**Comment by Tondas** [comment]
Hi @Doddler , 
do you know please, how to solve the issue with "Cannot find compilation library location for package".
Thanks
